File: Infrastructure/Entities/Project.cs
namespace Infrastructure.Entities
{
    using System;
    using System.Collections.Generic;

    public class Project
    {
        public Project()
        {
            this.ProjectAssignments = new List<ProjectAssignment>();
        }

        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public ProjectStatus Status { get; set; }

        public ICollection<ProjectAssignment> ProjectAssignments { get; set; }
    }
}

File: WebHost/ClientApp/src/app/models/Project.ts
import { AbstractControl } from '@angular/forms';

export class Project {

  public url: string;
  public name: string;
  public description: string;
  public startDate: Date;
  public endDate: Date;
  public status: string;
  public developerContextUrl:string;

  constructor() {


  }
  public static fromForm(form: AbstractControl): Project {
    let newProj = new Project();
    newProj.name = form.get('name').value;
    newProj.description = form.get('description').value;
    newProj.startDate = new Date(form.get('startDate').value);
    newProj.endDate = new Date(form.get('endDate').value);
    newProj.status = form.get('status').value;
    return newProj;
  }
}
